src/core/public/chrome/nav_links/to_nav_link.ts

```diff
@@ -24,25 +24,40 @@ import { appendAppPath } from '../../application/utils';
 
 export function toNavLink(app: App | LegacyApp, basePath: IBasePath): NavLinkWrapper {
   const useAppStatus = app.navLinkStatus === AppNavLinkStatus.default;
-  const baseUrl = isLegacyApp(app) ? basePath.prepend(app.appUrl) : basePath.prepend(app.appRoute!);
+  const relativeBaseUrl = isLegacyApp(app)
+    ? basePath.prepend(app.appUrl)
+    : basePath.prepend(app.appRoute!);
+  const url = relativeToAbsolute(appendAppPath(relativeBaseUrl, app.defaultPath));
+  const baseUrl = relativeToAbsolute(relativeBaseUrl);
+
   return new NavLinkWrapper({
     ...app,
     hidden: useAppStatus
       ? app.status === AppStatus.inaccessible
       : app.navLinkStatus === AppNavLinkStatus.hidden,
     disabled: useAppStatus ? false : app.navLinkStatus === AppNavLinkStatus.disabled,
     legacy: isLegacyApp(app),
-    baseUrl: relativeToAbsolute(baseUrl),
+    baseUrl,
     ...(isLegacyApp(app)
-      ? {}
+      ? {
+          href: url && !url.startsWith(app.subUrlBase!) ? url : baseUrl,
+        }
       : {
-          url: relativeToAbsolute(appendAppPath(baseUrl, app.defaultPath)),
+          href: url,
+          url,
         }),
```

Comment:
Mostly to align with EUI prop names otherwise it's one more prop I have to alias at the component level.

Also has a couple side benefits:
- Makes sure that nothing breaks if something else was low key relying on `url` being a certain thing at a certain time so makes the refactor a little safer (can refactor one thing at a time)
- Also makes clear what this will be used for (URLs can be constructed in a lot of different ways so knowing exactly what you're getting can be a bit of a toss up. With `href` you know exactly what the ultimate purpose of the prop is.)